Examples/Algorithms/AmbiguityResolution/src/GreedyAmbiguityResolutionAlgorithm.cpp

```diff
@@ -0,0 +1,101 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/AmbiguityResolution/GreedyAmbiguityResolutionAlgorithm.hpp"
+
+#include "Acts/AmbiguityResolution/GreedyAmbiguityResolution.hpp"
+#include "Acts/EventData/MultiTrajectoryHelpers.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/Measurement.hpp"
+#include "ActsExamples/Framework/ProcessCode.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+
+#include <cstddef>
+#include <iterator>
+#include <numeric>
+#include <stdexcept>
+
+#include <boost/container/flat_map.hpp>
+#include <boost/container/flat_set.hpp>
+
+namespace {
+
+Acts::GreedyAmbiguityResolution::Config transformConfig(
+    const ActsExamples::GreedyAmbiguityResolutionAlgorithm::Config& cfg) {
+  Acts::GreedyAmbiguityResolution::Config result;
+  result.maximumSharedHits = cfg.maximumSharedHits;
+  result.maximumIterations = cfg.maximumIterations;
+  result.nMeasurementsMin = cfg.nMeasurementsMin;
+  return result;
+}
+
+std::size_t sourceLinkHash(const Acts::SourceLink& a) {
+  return static_cast<std::size_t>(
+      a.get<ActsExamples::IndexSourceLink>().index());
+}
+
+bool sourceLinkEquality(const Acts::SourceLink& a, const Acts::SourceLink& b) {
```

Comment:
Just fyi, with this being in the CPP file, without LTO it's not getting inlined, which sort of negates the benefit of the template arguments.

Reply:
I thought it can still benefit from inlining because the call which receives the template arguments is also in this CPP file

Reply:
Yeah, maybe